### PR TITLE
Update Code Review/Pull Request Guidelines

### PR DESCRIPTION
**Pull request by zurbergram**
Proposing changes to the code review/pull request guidelines to help promote more efficient and productive code reviews.

- Using https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
- Use team specific label(s)
- not posting in platform-support for a VSP team review unless it’s been 24hrs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: I dig it! This is awesome.

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: Let's stick to 300 for now.

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: Probably doesn't need to be a sub-point of the team labels item.

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]:
```suggestion
  * Tag your PR with the following labels:
    * For *all code*, add your team-specific label(s)
    * For *database migrations*, add the `database-migration` label
    * For *dependency/package upgrades*, add the `package-change` label
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: I like the idea of going through the lead or scrum master, but I think it still ultimately needs to go through Leah or Patrick.

Does that sound right, @gunsch?

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: :100:

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: This _sounds_ right, but I'm not familiar with the process myself.

@annaswims What're your thoughts?

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: Does this label exist right now?

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: Is this how the `database-migration` label is being used right now?

**Review by cvalarida — COMMENTED** [review]
Overall, I think it looks good, but there are a few things worth calling out.

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [reply Practice Areas/Engineering/code_review_guidelines.md]: to leave the wording more general, let's say "escalate via your DSVA product owner"

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [reply Practice Areas/Engineering/code_review_guidelines.md]: cc @annaswims

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (annaswims) [reply Practice Areas/Engineering/code_review_guidelines.md]: This label doesn't exist in vets-api.  Can you explain the need for this label?

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (annaswims) [reply Practice Areas/Engineering/code_review_guidelines.md]: I think this is generally covered in `discuss how code changes are related to the pull request objectives`

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: changed

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: removed

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: Easier for reviewers to know at a glance when PRs contain database or package/dependency changes

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: The BAH team has started using it in https://github.com/department-of-veterans-affairs/gibct-data-service to allow reviewers to more easily know if a PR contains database changes

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: Created the labels for 
- https://github.com/department-of-veterans-affairs/gibct-data-service
- https://github.com/department-of-veterans-affairs/vets-api
- https://github.com/department-of-veterans-affairs/vets-website

Comment (gunsch) [reply Practice Areas/Engineering/code_review_guidelines.md]: Hmm, I don't think we need labels for that though. If a PR contains database or package changes, call that out in the issue heading and/or PR description?

I'll leave this up to @annaswims to make a final call on whether or not we want this, but in general we have a preference _against_ adding extra process until there's a clear need for it.

Comment (cvalarida) [reply Practice Areas/Engineering/code_review_guidelines.md]: I agree with @gunsch. I think this kind of indication can be added in the PR description as needed.

In general, PR descriptions should be pretty thorough, so it seems like a good fit there.

Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: Which would be better?
```
  * Tag your PR with the following labels:
    * For *all code*, add your team-specific label(s)
    * (Optional) For *database migrations*, add the `database-migration` label
    * (Optional) For *dependency/package upgrades*, add the `package-change` label
```
or
```
  * Tag your PR with the following labels:
    * For *all code*, add your team-specific label(s)
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: Might be worth calling out how to do this. Is there a button to graduate a draft PR to a full-fledged PR?

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (cvalarida) [inline Practice Areas/Engineering/code_review_guidelines.md]: I like this better, but I'm not sure if we should nix the bit about reverting the change if there was code that was added / changed after the approval.

@gunsch What do you think?

**Review by cvalarida — COMMENTED** [review]
I had a couple more questions, but I really like the direction this is taking.

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [reply Practice Areas/Engineering/code_review_guidelines.md]: Please leave the original wording. The point is to clearly indicate what action VSP team will take if non-trivial changes are submitted after review.

If you want to see that clarified better, I'd prefer:

```suggestion
    * If non-trivial changes to your code are made *after approval by VSP team* and submitted without a re-review, your change is subject to being reverted.
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: There is, 
Screen Shot 2019-08-07 at 2 19 41 PM

Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: updating section to indicate this

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: made suggested change

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: also added link to https://github.blog/2019-02-14-introducing-draft-pull-requests/

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]: let's drop this line, too -- line 14 already links to the other Draft PR doc, and line 25 has this link too

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]:
```suggestion
  * Tag your PR with your team-specific label(s)
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]:
```suggestion
  * For this first review, your code should be reviewed and approved by a member of your team. 
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]:
```suggestion
      * An engineer from the VSP team will aim to review within 24 hours (business days) and each additional round of changes will be reviewed within 24 hours.  If you need a review faster, escalate via your DSVA product owner.
```

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (gunsch) [inline Practice Areas/Engineering/code_review_guidelines.md]:
```suggestion
      * **Do not post in Slack channels requesting VSP code reviews** unless it has been more than 24 hours since requesting review on GitHub.
```

**Review by gunsch — COMMENTED** [review]
went through with a closer edit view, made a bunch of small comments but otherwise this LGTM

File: Practice Areas/Engineering/code_review_guidelines.md
Comment (zurbergram) [reply Practice Areas/Engineering/code_review_guidelines.md]: removed and changed link above to this link, as https://github.blog/2019-02-14-introducing-draft-pull-requests/ is more helpful with how to use draft pull requests

**Review by gunsch — APPROVED** [review]
LGTM

**Pull request merged**
